### PR TITLE
feat: extended session key plugin implementation

### PR DESCRIPTION
**Pull request by denniswon**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on extending the functionality of the SessionKeyPlugin in the accounts package. 

### Detailed summary
- Added `buildSessionKeysToRemoveStruct` function to build a struct of session keys to remove.
- Added `ExtendedSessionKeyPlugin` extension that includes additional methods for updating, adding, removing, and clearing account session keys.
- Imported necessary types and dependencies.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/accounts/src/msca/plugins/session-key/extension.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import type {
+  Address,
+  ISmartAccountProvider,
+  SupportedTransports,
+} from "@alchemy/aa-core";
+import type { IMSCA } from "../../types.js";
+import { SessionKeyPlugin } from "./plugin.js";
+import { buildSessionKeysToRemoveStruct } from "./utils.js";
+
+export const ExtendedSessionKeyPlugin = {
```
Comment (denniswon) [inline packages/accounts/src/msca/plugins/session-key/extension.ts:10]: other methods we should include here you can think of? @moldy530

Comment (moldy530) [reply packages/accounts/src/msca/plugins/session-key/extension.ts:10]: Once we have the latest ABIs we'll probably also want to support `rotateSessionKey`

File: packages/accounts/src/msca/plugins/session-key/utils.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// find predecessors for each keys and returned the struct `ISessionKeyPlugin.SessionKeyToRemove[]`
+
+import {
+  type Address,
+  type ISmartAccountProvider,
+  type ISmartContractAccount,
+} from "@alchemy/aa-core";
+import { SessionKeyPlugin } from "./plugin.js";
+
+// where SessionKeyToRemove = { sessionKey: Address, predecessor: Hex }
```
Comment (moldy530) [inline packages/accounts/src/msca/plugins/session-key/utils.ts:10]: think something happened here with the comments

**Review by moldy530 — APPROVED** [review]
Generally LGTM, we might need to add a few more once the new session key code is merged in. The new session key plugin introduces methods like:

`addSessionKey`, `removeSessionKey`, `rotateSessionKey` which operate on just one key

File: packages/accounts/src/msca/plugins/session-key/extension.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        const isSessionKey = await contract.read.isSessionKey([key], {
+          account: await provider.account.getAddress(),
+        });
+        return isSessionKey;
+      },
+    };
+  },
+};
+
+export const ExtendedSessionKeyPlugin: Plugin<
```
Comment (denniswon) [inline packages/accounts/src/msca/plugins/session-key/extension.ts:76]: @moldy530 strongly typed as in like this right?

**Pull request merged**